### ISSUE TITLE
Using Entitas with MonoGame, a few questions.

### ISSUE THREAD

**Issue by Substr** [sschmid/Entitas-CSharp#240]
Hi, I've been testing out using Entitas in a MonoGame project, but having a bit of trouble figuring out a few things.

1) What's the correct way to perform the initial setup of the pools? I came across [this example](https://github.com/jamiltron/BasicEntitasMonogameExample/blob/master/BasicEntitasMonogameExample/jamiltron/BEMEGame.cs) which seems to just immediately reference `Pools.pool` inside of `Initialize()` but this was undefined for me, possibly outdated example? I then found [your own basic c-sharp example](https://github.com/sschmid/EntitasPure/blob/master/EntitasPure/Program.cs) and used:

    `var pools = Pools.sharedInstance;` 
    `pools.SetAllPools();` 

    which seems to work, and I then have access to `pools.core`  etc.  Is this the correct way to init things?     I couldn't find anything on the docs about it exactly.

2) Is there some clean way I can pass along the deltaTime from MonoGame's `Update()` method to the `Execute()` call to the systems? I was hoping for an overload which accepted a parameter that was passed along to every system's own `Execute()` method. I'm unsure about Unity, but at least with MonoGame / XNA it's quite usual to need access to the delta inside of systems during the update / draw loops.

Thanks!

**Comment by sschmid** [comment]
Hi,

I recommend setting up the pools like this
```csharp
 var pools = Pools.sharedInstance; // lazy init of the pools instance
 pools.SetAllPools(); // Sets up all pool (e.g. core, meta, depending on your specified pools)
```

If you use the code generator it will extend the Pools class to contain all your specified pools (e.g. core, meta, ui, etc)

That's pretty much it as you can see in [Entitas Pure](https://github.com/sschmid/EntitasPure/blob/master/EntitasPure/Program.cs). Pass in the pools into systems via constructor argument (`ISetPools` is deprecated as of today) to keep things decoupled.

About delta time:
Entitas itself doesn't have the concept of time baked into the core, but it is really easy to add either deltaTime (DeltaTimeComponent) or frameCount (TickComponent)
In our [Unite Talk (watch)](https://www.youtube.com/watch?v=lNTaC-JWmdI) we discussed that for us time is the same as input.
I recommend having DeltaTimeComponent [SingleEntity] and do this (or similar)
```
var deltaTime = (float)gameTime.ElapsedGameTime.TotalSeconds;
pools.input.ReplaceDeltaTime(deltaTime);
```
in the beginning of mono game update.

This way every system can easily read the deltaTime `pools.input.deltaTime.value`

Hope that helps! :)

**Comment by sschmid** [comment]
This way you can also write unit tests based on deltaTime without depending on a running game. Time is input ;)

**Comment by Substr** [comment]
That helped for sure! It's a nice way to think of the timing as just another component as you said.

One other concept that I'm still not totally clear on is when you would use different pools vs. when you just fetch Groups using the Matchers from the one pool. In the example you gave there as well, you seemed to be accessing the entity attribute directly through the "input pool" or am I reading that wrong? Is that due to the [SingleEntity] attribute?

**Comment by Substr** [comment]
I seem to have solved the error above by adding [Core] attribute to the component, although I'm not at all clear why that solved it, or if you're then always supposed to specify one or more pools for all components like this?

**Comment by sschmid** [comment]
Each component is in the default pool when there is no PoolAttribute. This is fine if you only have one pool. It seems like you already use your custom pool called Core (which is great). All the components that should be available in this pool have to have the CoreAttribute.

A pool is basically like a context. Every component that should be available in this context needs the related PoolAttribute. Components can have multiple different PoolAttributes to be available in multiple pools.

**Comment by Substr** [comment]
@sschmid Thanks for clarifying. I seemed to have a problem with just using the default pool however, giving errors like the one above when trying to look up entities by those components. It seemed required to add a pool attribute to every component to make it work. I also found that once you add a pool attribute like "[Core]" you then can't use Matcher to find it but instead have to use CoreMatcher - this was a little confusing.
